src/MacroTools/Mechanics/StructurePacking.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using WCSharp.Events;
+using static War3Api.Common;
+
+namespace MacroTools.Mechanics
+{
+  /// <summary>
+  /// System to allow structures to convert themselves into units. Those units can then use an ability to rebuild the building.
+  /// </summary>
+  public class StructurePacking
+  {
+    /// <summary>
+    /// Object that contains information and methods for packable structures
+    /// </summary>
+    public class PackableStructure
```

Comment:
```suggestion
    public sealed class PackableStructure
```
I'm personally a fan of making classes sealed when they're not explicitly designed to be inheritable.